src/undoengine.js

```diff
@@ -65,6 +65,13 @@ export default class UndoEngine extends Plugin {
 
 		this.listenTo( this.editor.model, 'applyOperation', ( evt, args ) => {
 			const operation = args[ 0 ];
+
+			// Do not register batch if the operation is not a document operation.
+			// This prevents from creating empty undo steps, where all operations where non-document operations.
```

Comment:
Link to the bug is missing.

Reply:
Unless you consider this completely obvious that undo should not record these changes?

Reply:
Well, I consider everything obvious :). I'll add one more line of comment to make it more obvious.